snakers4_silero-models_stt.md

```diff
@@ -0,0 +1,77 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+category: researchers
+title: Silero Speech-To-Text Models
+summary: A set of compact enterprise-grade pre-trained STT Models for multiple languages.
+image: silero_logo.jpg
+author: Silero AI Team
+tags: [audio, scriptable]
+github-link: https://github.com/snakers4/silero-models
+github-id: snakers4/silero-models
+featured_image_1: silero_stt_model.jpg
+featured_image_2: silero_imagenet_moment.png
+accelerator: cuda-optional
+---
+
+```python
+import torch
+import zipfile
+import torchaudio
+from glob import glob
+# see https://github.com/snakers4/silero-models for utils and more examples
+
+device = torch.device('cpu')  # gpu also works, but our models are fast enough for CPU
+model, decoder, utils = torch.hub.load(github='snakers4/silero-models',
+                                       model='silero_stt',
+                                       device=device,
+                                       force_reload=True)
+(read_batch,
+ split_into_batches,
+ read_audio,
+ prepare_model_input) = utils  # see function signature for details
+
+torch.hub.download_url_to_file('http://www.openslr.org/resources/83/midlands_english_female.zip',
```

Comment:
instead of a zip file, extracting it and then running a batch of wav files through, it would be much nicer and more illustrative of downloading a single wav file and processing it through.

like:

```
torch.hub.download_url_to_file('some download url for speech.wav', dst='speech.wav)

input = prepare_model_input(glob('midlands_english_female/*.wav'), device=device)

output = model(input)
```

Reply:
one of reasons why I am downloading whole validation dataset and running a batch is to demonstrate the models are actually quite fast on CPU as well as on GPU

but I see your point 
I will add a default example with one file, but I would nevertheless keep the utils
because for a first time user batching may be an issue, so I would like to solve that

I will add the changes shortly